test/assignment.coffee

```diff
@@ -947,64 +947,41 @@ test "#4673: complex destructured object spread variables", ->
   eq g, 1
 
 test "#4878: Compile error when using destructuring with a splat or expansion in an array", ->
-  eqJS '''
-    foo = (list) ->
-      [first, ..., last] = list
-    ''', '''
-    var foo,
-      slice = [].slice;
-
-    foo = function(list) {
-      var first, last;
-      return [first] = list, [last] = slice.call(list, -1);
-    };
-    '''
-
-  eqJS '''
-    foo = (list) ->
-      [first..., last] = list
-    ''', '''
-    var foo,
-      splice = [].splice;
-
-    foo = function(list) {
-      var first, last;
-      return [...first] = list, [last] = splice.call(first, -1);
-    };
-    '''
-
-  eqJS '''
-    foo = (list) ->
-      ret =
-        if value.length
-          [ first, ..., last ] = value
-          [ first, last ]
-        else
-          []
-    ''', '''
-    var foo,
-      slice = [].slice;
-
-    foo = function(list) {
-      var first, last, ret;
-      return ret = value.length ? (([first] = value, [last] = slice.call(value, -1)), [first, last]) : [];
-    };
-    '''
-
-  eqJS '''
-    foo = (list) ->
-      ret =
-        if value.length
-          [ first..., last ] = value
-          [ first, last ]
-        else
-          []
-    ''', '''
-    var foo,
-      splice = [].splice;
-
-    foo = function(list) {
-      var first, last, ret;
-      return ret = value.length ? (([...first] = value, [last] = splice.call(first, -1)), [first, last]) : [];
-    };
-    '''
+  arr = ['a', 'b', 'c', 'd']
+
+  f1 = (list) ->
+    [first, ..., last] = list
+
+  f2 = (list) ->
+    [first..., last] = list
+
+  f3 = (list) ->
+    ([first, ...] = list); first
+
+  f4 = (list) ->
+    ([first, ...rest] = list); rest
+
+  arrayEq f1(arr), ['d']
```

Comment:
In 2.1.1, `f1(arr)` returns `[ 'a', 'b', 'c', 'd' ]`

Reply:
In 2.1.1 this
```coffeescript
f1 = (list) ->
 [first, ..., last] = list
```

compiles into 

```javascript
f1 = function(list) {
  var first, last;
  return first = list[0], last = list[list.length - 1], list;
};
```

So, the returned value is `list`.

In 2.2.0 it's compiled to

```javascript
f1 = function(list) {
  var first, last;
  return [first] = list, [last] = slice.call(list, -1);
};
```

The returned value is `[last]'.

Reply:
Yes, but `[first, ..., last] = list` is shorthand for `return [first, ..., last] = list`, which looks to me like it should return the full array. As in, returning just `[last]` is wrong.

```coffee
do -> [a, b, c] = [1, 2, 3]
# returns [1, 2, 3]
```

Reply:
Yup. Sorry, my mistake. Not enogh coffe in the system :sleepy:

Reply:
> Not enough coffee in the system

Isn’t that why you’re here? 😝